module/sidebar/components/menu-component.js

```diff
@@ -113,22 +113,73 @@ export default class MenuComponent extends Component {
             const target = game.user?.targets?.first();
 
             let allowed = "true";
-            
+            let args = { accuracyModifier: 0, captureRateModifier: 0 };
+
+            // Check if Alt key is pressed
+            if (event.altKey) {
+                const dialogPromise = new Promise((resolve, reject) => {
+                    // Display prompt dialog to get custom accuracy and capture rate modifiers                
+                    const d = new Dialog({
+                        title: "Throw Pokeball with Modifiers",
+                        content: `
+                            <form>
+                                <div class="form-group">
+                                    <label>Accuracy Modifier</label>
+                                    <input type="number" name="accuracyModifier" value="0">
+                                </div>
+                                <div class="form-group">
+                                    <label>Capture Rate Modifier</label>
+                                    <input type="number" name="captureRateModifier" value="0">
+                                </div>
+                            </form>
+                        `,
+                        buttons: {
+                            throw: {
+                                icon: '<i class="fas fa-dice-d20"></i>',
+                                label: "Throw",
+                                callback: (html) => {
+                                    args.accuracyModifier = parseInt(html.find('[name="accuracyModifier"]').val());
+                                    args.captureRateModifier = parseInt(html.find('[name="captureRateModifier"]').val());
+                                    resolve(args);
+                                },
+                            },
+                            cancel: {
+                                icon: '<i class="fas fa-times"></i>',
+                                label: "Cancel",
+                                callback: () => reject("User cancelled the dialog"),
+                            }
+                        },
+                        default: "throw",
+                        close: () => reject("User closed the dialog without submitting"),
+                    })
+
+                    d.render(true);
+                });
+
+                await dialogPromise
+                    .then((args) => {
+                        console.log(args);
+                    })
+                    .catch((error) => {
+                        console.log(error);
+                    });
```

Comment:
In this section of code you're doing a lot of editing out-of-scope, while in this case it works, it's not proper to edit the `args` like this, nor calling the await as such.

Instead I'd change the callbacks to be something like:
```js
callback: (html) => {
    const args = {
        accuracyModifier: parseInt(html.find('[name="accuracyModifier"]').val()),
        captureRateModifier: parseInt(html.find('[name="captureRateModifier"]').val())
    }
    resolve(args)
}
```
which can even be simplified to `resolve({/*params*/})`

Then for the rejects you can do the logging inside the callback
```js
callback: () => {
    console.log("User cancelled the dialog");
    reject();
}
```
Meanwhile returning `undefined`.

Then at the top we can simply call
`const args = await new Promise(/*etc.*/);`

As in this file there are no references to args.property we don't have to do a null check here and instead can perform that in the `throwPokeball` or `throwPokeballRequest` methods, where either null checks or null coalescing can be used.